packages/mongo/collection.js

```diff
@@ -301,6 +298,33 @@ Object.assign(Mongo.Collection.prototype, {
     }
   },
 
+  // The Mongo Node driver is configured to `ignoreUndefined`
+  // fields in documents/selectors. This function can be used to remove
+  // `undefined` fields, to keep things consistent, and help avoid any
+  // unwarranted side effects (such as issue #9619).
+  _removeUndefinedFields(selector) {
+    if (typeof selector !== 'object') {
+      return selector;
+    }
+
+    const selectorCopy = EJSON.clone(selector);
+    const seen = new Set();
+    (function removeUndefinedRecursively(cleanSelector) {
+      if (!seen.has(cleanSelector)) {
+        seen.add(cleanSelector);
+        Object.keys(cleanSelector).forEach((key) => {
+          if (cleanSelector[key] && typeof cleanSelector[key] === 'object') {
+            removeUndefinedRecursively(cleanSelector[key]);
+          } else if (typeof cleanSelector[key] === 'undefined') {
+            delete cleanSelector[key];
```

Comment:
This may be tricky, but I wonder if there's a way to avoid the `EJSON.clone` if we aren't going to delete any fields from the original `selector`.